### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-python-library.git`](https://github.com/cisagov/skeleton-python-library.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/guacscanner.git guacscanner
    cd guacscanner
    git remote add skeleton https://github.com/cisagov/skeleton-python-library.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .github/workflows/build.yml README.md setup.py src/guacscanner/_version.py 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may *append* to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

---

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

File: src/guacscanner/guacscanner.py [hunk 1]
```diff
@@ -51,6 +51,7 @@
 import string
 import sys
 import time
+import typing
```
Comment (mcdonnnj) [inline src/guacscanner/guacscanner.py]: We have historically imported only the pieces of this library that we are using.

```suggestion
from typing import Optional
```

File: src/guacscanner/guacscanner.py [hunk 2]
```diff
@@ -278,8 +279,8 @@ def get_entity_id(db_connection, entity_name, entity_type):
 def add_user(
     db_connection: psycopg.Connection,
     username: str,
-    password: str = None,
-    salt: bytes = None,
+    password: typing.Optional[str] = None,
+    salt: typing.Optional[bytes] = None,
```
Comment (mcdonnnj) [inline src/guacscanner/guacscanner.py]:
```suggestion
    password: Optional[str] = None,
    salt: Optional[bytes] = None,
```

**Review by mcdonnnj — APPROVED** [review]
Two suggestions around a typing addition for your consideration.

File: src/guacscanner/guacscanner.py (hunk 1 above)
Comment (jsf9k) [reply src/guacscanner/guacscanner.py]: In this case both `typing.Optional` and `schema.Optional` are floating around and I can't have them clobbering each other.

File: src/guacscanner/guacscanner.py (hunk 2 above)
Comment (jsf9k) [reply src/guacscanner/guacscanner.py]: In this case both `typing.Optional` and `schema.Optional` are floating around and I can't have them clobbering each other.

File: src/guacscanner/guacscanner.py (hunk 1 above)
Comment (mcdonnnj) [reply src/guacscanner/guacscanner.py]: I would prefer we switch the `schema` import to `import schema` and prefix the pieces we use and leave the `from typing import Optional`. The `typing` imports are more likely to be interspersed in the code base while the `schema` stuff will be isolated to where we do argument validation.

File: src/guacscanner/guacscanner.py (hunk 2 above)
Comment (mcdonnnj) [reply src/guacscanner/guacscanner.py]: I would prefer we switch the `schema` import to `import schema` and prefix the pieces we use and leave the `from typing import Optional`. The `typing` imports are more likely to be interspersed in the code base while the `schema` stuff will be isolated to where we do argument validation.

Comment (jsf9k) [reply src/guacscanner/guacscanner.py]: Please see comment 0b35318.

File: src/guacscanner/guacscanner.py (hunk 1 above)
Comment (jsf9k) [reply src/guacscanner/guacscanner.py]: Please see comment 0b35318.

**Review by dav3r — APPROVED** [review]
👍

File: src/guacscanner/guacscanner.py [hunk 3]
```diff
@@ -682,24 +683,24 @@ def main() -> None:
     # Parse command line arguments
     args = docopt.docopt(__doc__, version=__version__)
     # Validate and convert arguments as needed
-    schema = Schema(
+    mySchema = schema.Schema(
```
Comment (mcdonnnj) [inline src/guacscanner/guacscanner.py]: Nit-picky but Python conventions prefers snake case variable names per [PEP8](https://peps.python.org/pep-0008/#function-and-variable-names)

```suggestion
    my_schema = schema.Schema(
```

File: src/guacscanner/guacscanner.py [hunk 4]
```diff
@@ -708,8 +709,8 @@ def main() -> None:
         }
     )
     try:
-        validated_args = schema.validate(args)
-    except SchemaError as err:
+        validated_args = mySchema.validate(args)
```
Comment (mcdonnnj) [inline src/guacscanner/guacscanner.py]: Paired with the above suggestion.

```suggestion
        validated_args = my_schema.validate(args)
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: src/guacscanner/guacscanner.py (hunk 1 above)
Comment (mcdonnnj) [reply src/guacscanner/guacscanner.py]: Thanks!

File: src/guacscanner/guacscanner.py (hunk 2 above)
Comment (mcdonnnj) [reply src/guacscanner/guacscanner.py]: Thanks!

**Review by mcdonnnj — APPROVED** [review]
:shipit:

**Pull request merged**